Sources/SpeziLLMLocalDownload/LLMLocalDownloadManager.swift

```diff
@@ -41,41 +42,73 @@ public final class LLMLocalDownloadManager: NSObject {
         }
     }
     
-    /// The delegate handling the download manager tasks.
-    @ObservationIgnored private var downloadDelegate: LLMLocalDownloadManagerDelegate?  // swiftlint:disable:this weak_delegate
     /// The `URLSessionDownloadTask` that handles the download of the model.
-    @ObservationIgnored private var downloadTask: URLSessionDownloadTask?
-    /// Remote `URL` from where the LLM file should be downloaded.
-    private let llmDownloadUrl: URL
-    /// Local `URL` where the downloaded model is stored.
-    let llmStorageUrl: URL
+    @ObservationIgnored private var downloadTask: Task<(), Never>?
     /// Indicates the current state of the ``LLMLocalDownloadManager``.
     @MainActor public var state: DownloadState = .idle
+    private let modelConfiguration: ModelConfiguration
     
+    @ObservationIgnored public var modelExists: Bool {
+        let repo = Hub.Repo(id: modelConfiguration.name)
+        let url = HubApi.shared.localRepoLocation(repo)
+        let modelFileExtension = ".safetensors"
+        
+        do {
+            let contents = try FileManager.default.contentsOfDirectory(atPath: url.path())
+            return contents.first(where: { $0.hasSuffix(modelFileExtension) }) != nil
+        } catch {
+            return false
+        }
+    }
     
     /// Creates a ``LLMLocalDownloadManager`` that helps with downloading LLM files from remote servers.
     ///
     /// - Parameters:
-    ///   - llmDownloadUrl: The remote `URL` from where the LLM file should be downloaded.
-    ///   - llmStorageUrl: The local `URL` where the LLM file should be stored.
-    public init(
-        llmDownloadUrl: URL = LLMUrlDefaults.llama2ChatModelUrl,
-        llmStorageUrl: URL = .cachesDirectory.appending(path: "llm.gguf")
-    ) {
-        self.llmDownloadUrl = llmDownloadUrl
-        self.llmStorageUrl = llmStorageUrl
+    ///   - modelConfiguration: TODO
```

Comment:
Please ensure to address all TODOs and all new elements are properly documented 👍